### ISSUE TITLE
[Bitwarden] Caching

### ISSUE THREAD

**Issue by glekner** [raycast/extensions#4351]
### Extension

https://www.raycast.com/jomifepe/bitwarden

### Description

Right now each time one opens the Bitwarden "Search Vault" Command, it fetches the entire vault.
Can we implement caching? (Ideally the user can choose the expiration time)

### Who will benefit from this feature?

All the users, for instant access to the vault after initial fetching.

**Comment by pomdtr** [comment]
The caching is implemented on the bitwarden cli side. 

We will not implement another cache on the extension side since it would mean storing your entire vault on raycast side, which would be a security concern.

**Comment by glekner** [comment]
@pomdtr I see.
Can you explain the security risk?
To my understanding, different extensions are not able to access the same cache, so it would only be available to the Bitwarden extension. We could also use the existing secrets for encryption of the cache.

**Comment by pomdtr** [comment]
The extension is just a thin wrapper around the cli, we only want to add the minimal amount of logic over it. 

If we added a cache implemented in raycast, the extension users would need to rely on another third party (raycast). I prefer to avoid this.

**Comment by pomdtr** [comment]
Thanks for your comprehension !
If you are annoyed by this high delay, you can create an issue on the bitwarden repo :)

**Comment by glekner** [comment]
sure @pomdtr 

FYI this was already done in an alfred extension
<img width="788" alt="image" src="https://user-images.githubusercontent.com/24938324/213740588-ffa1a0e4-1454-4603-9fb2-27e0fbb5db77.png">

**Comment by pomdtr** [comment]
Interesting approach, it might be a good middle ground.

**Comment by pomdtr** [comment]
Btw I'm not the maintainer of the extension anymore, @jomifepe took the role. I was just giving my feedback here.

**Comment by jomifepe** [comment]
I agree with @pomdtr regarding the security of caching secrets, it's better to be safe than sorry when dealing with password managers and risking creating an exploit.

> sure @pomdtr cc @bengry
> 
> FYI this was already done in an alfred extension https://github.com/blacs30/bitwarden-alfred-workflow
> 
> <img alt="image" width="788" src="https://user-images.githubusercontent.com/24938324/213740588-ffa1a0e4-1454-4603-9fb2-27e0fbb5db77.png">

Yeah, that might be a good approach to get a faster vault list, I can research this.

**Comment by verheyenkoen** [comment]
Anyone can pick this up or point? I can also try to give it a shot if someone can point me in the right direction (where to hook, how to cache, does raycast have a built-in caching mechanism,...).

**Comment by pomdtr** [comment]
I think the approach of the Alfred extension is fine (behind a setting that the user would manually set), but wait for @jomifepe approval before starting any work

**Comment by jomifepe** [comment]
> Anyone can pick this up? I can also try to give it a shot if someone can point me in the right direction (where to hook, how to cache, does raycast have a built-in caching mechanism,...).

Hey, I have picked this up, just haven't finished yet. Yes, Raycast does have caching mechanisms, but because we try to avoid caching sensitive values, the challenge has been to make sure these values are available when accessing them.

**Comment by jomifepe** [comment]
Caching is now implemented, try it out in the latest version.

**Comment by jomifepe** [comment]
@raycastbot close this issue
